core/src/main/java/com/transferwise/common/gracefulshutdown/strategies/TaskSchedulersGracefulShutdownStrategy.java

```diff
@@ -0,0 +1,72 @@
+package com.transferwise.common.gracefulshutdown.strategies;
+
+import com.transferwise.common.gracefulshutdown.GracefulShutdownStrategy;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.concurrent.ConcurrentTaskScheduler;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+
+@Slf4j
+public class TaskSchedulersGracefulShutdownStrategy implements GracefulShutdownStrategy {
+
+  @Autowired
+  private ApplicationContext applicationContext;
+
+  private AtomicInteger inProgressShutdowns = new AtomicInteger();
+
+  @Override
+  public void prepareForShutdown() {
+    var taskSchedulers = applicationContext.getBeansOfType(TaskScheduler.class).values();
+
+    var executors = Executors.newFixedThreadPool(10);
+    for (var taskScheduler : taskSchedulers) {
+      inProgressShutdowns.incrementAndGet();
```

Comment:
should this be initialised with `taskSchedulers.size()`? 

Could it be that can `canShutdown()` returns true if invoked right after the first task scheduler get closed?

Reply:
This can not happen, but for better readability, I changed to .size().